repository/kafka/docs/latest/kudo-kafka-runbook.md

```diff
@@ -0,0 +1,128 @@
+# KUDO Kafka Runbook
+
+## Overview
+
+This runbook helps the user to perform a demonstration to experience user workload generation feature offered by KUDO Kafka.
+
+## Getting Started
+
+To experience KUDO Kafka with some user workload on its brokers, some prerequisites are:
+- A running K8s cluster (with at least 3 worker nodes)
+- KUDO initialized and configured over the `kubectl`
+
+### Creating a namespace
+To isolate our operators from other running installations over the cluster, we prefer to create a separate namespace for them.
+
+Let us create a namespace `kudo-kafka` using YAML definition `kudo-kafka-ns.yaml` defined as:
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kudo-kafka
+```
+After creating namespace definition file `kudo-kafka-ns.yaml`, we need to create namespace over K8s cluster as:
+```
+kubectl apply -f kudo-kafka-ns.yaml
+```
+The output of this command should be something like:
+        `namespace/kudo-kafka created`
+
+## Deploying Zookeeper
+
+KUDO Kafka should be backed by KUDO Zookeeper running in the background. To install KUDO Zookeeper in the namespace `kudo-kafka`:
+```
+kubectl kudo install zookeeper --instance=zk --namespace=kudo-kafka
+```
+
+The output of this command should be something like:
+```
+operator.kudo.dev/v1beta1/zookeeper created
+operatorversion.kudo.dev/v1beta1/zookeeper-0.2.0 created
+instance.kudo.dev/v1beta1/zk created
+```
+
+You can check for the successful deployment of Zookeeper service through: 
+```sh
+$ kubectl kudo plan status --instance=zk --namespace=kudo-kafka
+Plan(s) for "zk" in namespace "kudo-kafka":
+.
+└── zk (Operator-Version: "zookeeper-0.2.0" Active-Plan: "deploy")
+    ├── Plan deploy (serial strategy) [COMPLETE]
+    │   ├── Phase zookeeper [COMPLETE]
+    │   │   └── Step deploy (COMPLETE)
+    │   └── Phase validation [COMPLETE]
+    │       ├── Step validation (COMPLETE)
+    │       └── Step cleanup (COMPLETE)
+    └── Plan validation (serial strategy) [NOT ACTIVE]
+        └── Phase connection (serial strategy) [NOT ACTIVE]
+            └── Step connection (serial strategy) [NOT ACTIVE]
+                ├── connection [NOT ACTIVE]
+                └── cleanup [NOT ACTIVE]
+```
+
+Please note that the deployment of Zookeeper will result in a Zookeeper URI address, 
+which will be further used to specify about Zookeeper's instance during KUDO Kafka's installation. The URI in `kudo-kafka` namespace can be referred as:
+  `zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181`
+
+### Deploying KUDO Kafka with dummy user workload
+
+KUDO Kafka can be deployed with some dummy user workload running over Kafka brokers by specifying `RUN_USER_WORKLOAD` parameter as true. 
+
+In order to install KUDO Kafka having user workload in the namespace `kudo-kafka`:
+```
+kubectl kudo install kafka --instance=kafka --namespace=kudo-kafka \
```

Comment:
remove from here all unnecessary parameters that already have sane defaults
```
kubectl kudo install kafka --instance=kafka --namespace=kudo-kafka \
 -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
 -p RUN_USER_WORKLOAD=true \
 -p ADD_SERVICE_MONITOR=true
```
should be enough for this runbook